utils/clients/gemini/client_test.go

```diff
@@ -0,0 +1,136 @@
+
+package gemini
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/brave-intl/bat-go/settlement"
+	"github.com/brave-intl/bat-go/utils/cryptography"
+	uuid "github.com/satori/go.uuid"
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/suite"
+)
+
+type GeminiTestSuite struct {
+	suite.Suite
+	secret cryptography.HMACKey
+	apikey string
+}
+
+func TestGeminiTestSuite(t *testing.T) {
+	suite.Run(t, new(GeminiTestSuite))
+}
+
+func (suite *GeminiTestSuite) SetupTest() {
+	secret := os.Getenv("GEMINI_CLIENT_SECRET")
+	apikey := os.Getenv("GEMINI_CLIENT_KEY")
+	if secret != "" {
+		suite.secret = cryptography.NewHMACHasher([]byte(secret))
+		suite.apikey = apikey
+	}
+}
+
+func (suite *GeminiTestSuite) TestBulkPay() {
+	ctx := context.Background()
+	client, err := New()
+	suite.Require().NoError(err, "Must be able to correctly initialize the client")
+	five := decimal.NewFromFloat(5)
+
+	// // uncomment if using a master key
+	// accountListRequest := suite.preparePrivateRequest(NewAccountListPayload())
+	// accounts, err := client.FetchAccountList(ctx, suite.apikey, suite.secret, accountListRequest)
+	// suite.Require().NoError(err, "should not error during account list fetching")
+	// primary := "primary"
+	// account := findAccountByClass(accounts, primary)
+	// suite.Require().Equal(primary, account.Class, "should have a primary account")
+
+	balancesRequest := suite.preparePrivateRequest(NewBalancesPayload())
+	balances, err := client.FetchBalances(ctx, suite.apikey, suite.secret, balancesRequest)
+	suite.Require().NoError(err, "should not error during balances fetching")
+	balance := findBalanceByCurrency(balances, "BAT")
+	suite.Require().True(
+		balance.Available.GreaterThanOrEqual(five),
+		"must have at least 5 bat to pass the rest of the test",
+	)
+
+	tx := settlement.Transaction{
+		// use this settlement id to create an ephemeral test
+		// SettlementID: uuid.NewV4().String(),
+		SettlementID: uuid.Must(uuid.FromString("4077459f-7389-46d3-a0d8-b1e56b2d279b")).String(),
+		Destination:  os.Getenv("GEMINI_TEST_DESTINATION_ID"),
+		Channel:      "brave.com",
+	}
+	BAT := "BAT"
+	payouts := []PayoutPayload{{
+		TxRef:       GenerateTxRef(&tx),
+		Amount:      five,
+		Currency:    BAT,
+		Destination: tx.Destination,
+	}}
+	bulkPayoutRequest := suite.preparePrivateRequest(NewBulkPayoutPayload(
+		os.Getenv("GEMINI_CLIENT_ID"),
+		&payouts,
+	))
+
+	_, err = client.UploadBulkPayout(ctx, suite.apikey, suite.secret, bulkPayoutRequest)
+	suite.Require().NoError(err, "should not error during bulk payout uploading")
+
+	pendingStatus := "Pending"
+	expectedPayoutResult := []PayoutResult{{
+		Result:      "OK",
+		TxRef:       GenerateTxRef(&tx),
+		Amount:      &five,
+		Currency:    &BAT,
+		Destination: &tx.Destination,
+		Status:      &pendingStatus,
+	}}
+	// suite.Require().Equal(&expectedPayoutResult, bulkPayoutResponse, "response should be predictable")
```

Comment:
is this needed anymore?

Reply:
can remove. was thinking it could be useful if the worker was to be run within seconds. it is not (10 mins)

Reply:
fixed in 9cd17a8